### PR TITLE
Progress Bar: Add Indeterminate state

### LINKED ISSUE #2785: Progress bar: Indeterminate state
### Prerequisites

- [X] I have [searched](https://github.com/trimble-oss/modus-web-components/issues?q=is%3Aissue) for duplicate or closed feature requests
- [X] I have read the [Modus style guidelines](https://modus.trimble.com/components/web/introduction/)

### Proposal

It will be nice have Indeterminate progress bar option in React like angular material has:
https://material.angular.io/components/progress-bar/overview

### Motivation and context

_No response_

### What is the issue regarding ?

@trimble-oss/modus-web-components

### Priority

Medium

### What product/project are you using Modus Components for ?

Numark

### What is your team/division name ?

Numark

### Are you willing to contribute ?

None

### Are you using Modus Web Components in production ?

_No response_

### PR DESCRIPTION
**Pull request by ElishaSamPeterPrabhu**
## Description

- Added prop called `mode` to determine determinate or indeterminate state
- Set aria-valuemax, aria-valuemin ,aria-valuenow to null

References 
Fixes #2785 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: stencil-workspace/src/components/modus-progress-bar/modus-progress-bar.tsx [hunk 1]
```diff
@@ -1,4 +1,3 @@
-// eslint-disable-next-line
```
Comment (coliff) [inline stencil-workspace/src/components/modus-progress-bar/modus-progress-bar.tsx:1]: this is needed at the top

File: stencil-workspace/src/components/modus-progress-bar/modus-progress-bar.scss [hunk 2]
```diff
@@ -26,6 +28,15 @@
     &.default-text-color {
       color: $col_white;
     }
+
+    &.indeterminate {
```
Comment (coliff) [inline stencil-workspace/src/components/modus-progress-bar/modus-progress-bar.scss:32]: please run css lint before committing so it auto-sorts and formats the CSS

File: stencil-workspace/src/components/modus-progress-bar/modus-progress-bar.tsx [hunk 3]
```diff
@@ -34,6 +33,9 @@ export class ModusProgressBar {
   /** (optional) The progress bar's value. */
   @Prop() value = 0;
 
+  /** (optional) The progress bar's mode. */
+  @Prop() mode: 'determinate' | 'indeterminate' = 'determinate';
```
Comment (prashanth-offcl) [inline stencil-workspace/src/components/modus-progress-bar/modus-progress-bar.tsx:37]: @ElishaSamPeterPrabhu Stencil style guide recommends to keep the props in alphabetical order 
https://stenciljs.com/docs/style-guide#high-level-example-commented

**Review by prashanth-offcl — CHANGES_REQUESTED** [review]
@ElishaSamPeterPrabhu please add E2E tests for the changes and resolve the build error and review comments.

File: stencil-workspace/src/components/modus-progress-bar/modus-progress-bar.tsx (hunk 1 above)
Comment (ElishaSamPeterPrabhu) [reply stencil-workspace/src/components/modus-progress-bar/modus-progress-bar.tsx:1]: added

File: stencil-workspace/src/components/modus-progress-bar/modus-progress-bar.tsx (hunk 3 above)
Comment (ElishaSamPeterPrabhu) [reply stencil-workspace/src/components/modus-progress-bar/modus-progress-bar.tsx:37]: Modified the props to be in aphabetical order

File: stencil-workspace/src/components/modus-progress-bar/modus-progress-bar.scss (hunk 2 above)
Comment (ElishaSamPeterPrabhu) [reply stencil-workspace/src/components/modus-progress-bar/modus-progress-bar.scss:32]: Ran lint test and modified css properties positions

**Comment by ElishaSamPeterPrabhu** [comment]
> @ElishaSamPeterPrabhu please add E2E tests for the changes and resolve the build error and review comments.

Added test cases and resolved comments

File: stencil-workspace/storybook/stories/components/modus-progress-bar/modus-progress-bar-storybook-docs.mdx [hunk 4]
```diff
@@ -26,6 +26,12 @@
...
 
+### Indeterminate
+
+<div>
+  <modus-progress-bar mode="indeterminate"></modus-progress-bar>
+</div>
+
 <Anchor storyId="components-progress-bar--compact" />
 
 ```html
```
Comment (prashanth-offcl) [inline stencil-workspace/storybook/stories/components/modus-progress-bar/modus-progress-bar-storybook-docs.mdx:37]: Can you add the example code for the indeterminate mode?

**Review by prashanth-offcl — APPROVED** [review]

**Pull request merged**